process.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
```

Comment:
Well, rendering templates aren't always pure disk IO - there's often network IO that blocks for far longer. So running these in parallel could be a big win.

I didn't realize this wasn't called with `--input-dir` - I haven't had a chance to look _too_ closely yet 😉. I'd definitely rather all separate files be rendered in parallel, whether or not they come from `--file` or `--input-dir`.